README.md
Hemingway Metro
======================
   Contributors: ddmboss designs  
   Tested up to: 3.9.1  
   Stable tag: 3.6  
   License: GPLv2 or later  
   License URI: http://www.gnu.org/licenses/gpl-2.0.html  

Description
==============
Hemingway Metro is based on the Hemingway Rewritten theme, that was created by Anders Noren and edited by Wordpress.com. This version has been further customized mainly in the home and archive pages.

Installation
================
1. In your admin panel, go to Appearance -> Themes and click the Add New button.
2. Click Upload and Choose File, then select the theme's .zip file. Click Install Now.
3. Click Activate to use your new theme right away.

Frequently Asked Questions
===========================
* Where can I add widgets?

Hemingway Metro includes four optional widget areas: a sidebar alongside your content, and three sidebars in the footer.

Configure these areas by going to Appearance → Widgets in your Dashboard.

* Does Hemingway Metro use featured images? 

Featured images are used on the archives, search and homepages, as well as the slider

* Does Hemingway Metro have social links? =

You can add links to a multitude of social services by following these steps:

1. Create a new Custom Menu, and assign it to the Social Links Menu location
2. Add links to each of your social services using the Links panel
3. Icons for your social links will automatically appear under the heart icon in the header

Quick Specs (all measurements in pixels)
====================
1. The main column width is 676.
2. The Primary Sidebar is 286.
3. The Footer column widths will adjust depending on the number of active sidebars:  
  a. 1 active = 1040  
  b. 2 active = 489 each  
  c. 3 active = 312 each  

Changelog
==========
* 1.0.8 - May 15, 2021 - Updated names and links
* 1.0.7 - August 18, 2014 - Featured Image fix
* 1.0.6 - August 15, 2014 - Homepage redesigned & Bug Fixes
* 1.0.5 - August 8, 2014 - More Bug Fixes
* 1.0.4 - August 7, 2014 - Bug Fixes
* 1.0.3 - August 7, 2014 - Changed the Homepage Thumbnails
* 1.0.2 - July 31, 2014 - Made the theme more efficient
* 1.0.1 - July 8, 2014 - Ready For Deployment
* 1.0.0 - June 30, 2014 - Begin Development

CHANGES.md
Hemingway Metro Changes Log
==========================
This is the official changes log for the hemingway metro theme

### 1.0.8 - May 15, 2021
Fixed and removed references to deprecated links
### 1.0.7 - August 18, 2014
Fixed Post thumbnail bugs
* Removed the masonry-init js file from theme
* Fixed a bug with posts showing multiple featured images
* Cleaned up template code

### 1.0.6 - August 15, 2014
Homepage Redesigned and A few bugs fixes
* Removed the custom homepage, theme now uses the default content layout
* Minor adjustments to the flexslider call code
* Removed custom styles and masonry from theme
* Changed the image sizes
* Theme now supports infinite scrolling again

### 1.0.5 - August 8, 2014
More Bug Fixes
* Moved Flexslider initializer to its own js file in the flexslider folder
* Changed the order js files are loaded in the functions file
* Added an official changelog file to the theme
* Removed content-media php file

### 1.0.4 - August 7, 2014
Bug Fixes
* Moved hemingway-metro js file back to the footer
* Moved masonry init script to its own js file

### 1.0.3 - August 7, 2014
Changed the homepage thumbnails to a masonry layout
* Thumbnails now scale to the width of the column instead of being cropped
* Moved hemingway-metro js file to the header

### 1.0.2 - July 31, 2014
Grouped JS files to speed up loading
* Moved Masonry-init and the flexslider initialize script into hemingway-metro js file
* Moved styles in the metro css file to the main stylesheet

### 1.0.1 - July 8, 2014
First Version is released
* This theme is a fork of the Hemingway Rewritten wordpress theme, this version has been edited, mainly on the homepage and archive pages, this theme is licensed under GPL 2+ so use it to create something awesome

### 1.0.0 - June 30, 2014
Theme went into development
* Forks Hemingway Rewritten theme
* Adds Custom Homepage layout
* Changed the Image sizes for the theme
* Added custom slider
* Removed all the Hemingway Rewritten tags from the theme


functions.php
<?php
/**
 * Hemingway Metro functions and definitions
 *
 * @package Hemingway Metro
 */

/**
 * Set the content width based on the theme's design and stylesheet.
 */
if ( ! isset( $content_width ) ) {
	$content_width = 676; /* pixels */
}

if ( ! function_exists( 'hemingway_metro_setup' ) ) :
/**
 * Sets up theme defaults and registers support for various WordPress features.
 *
 * Note that this function is hooked into the after_setup_theme hook, which
 * runs before the init hook. The init hook is too late for some features, such
 * as indicating support for post thumbnails.
 */
function hemingway_metro_setup() {

	/*
	 * Make theme available for translation.
	 * Translations can be filed in the /languages/ directory.
	 * If you're building a theme based on Hemingway Metro, use a find and replace
	 * to change 'hemingway-metro' to the name of your theme in all the template files
	 */
	load_theme_textdomain( 'hemingway-metro', get_template_directory() . '/languages' );

	// Add default posts and comments RSS feed links to head.
	add_theme_support( 'automatic-feed-links' );

	/*
	 * Enable support for Post Thumbnails on posts and pages.
	 *
	 * @link http://codex.wordpress.org/Function_Reference/add_theme_support#Post_Thumbnails
	 */
	add_theme_support( 'post-thumbnails' );
    add_image_size( 'hemingway-metro-header', '1280', '416', true );
    add_image_size( 'hemingway-metro-slider', '1040', '446', true );
	add_image_size( 'hemingway-metro-featured', '676', '290', true );

	// This theme uses wp_nav_menu() in one location.
	register_nav_menus( array(
		'primary' => __( 'Primary Menu', 'hemingway-metro' ),
	) );

	// Enable support for Post Formats.
	add_theme_support( 'post-formats', array( 'aside', 'audio', 'chat', 'gallery', 'image', 'link', 'quote', 'status', 'video' ) );

	// Setup the WordPress core custom background feature.
	add_theme_support( 'custom-background', apply_filters( 'hemingway_metro_custom_background_args', array(
		'default-color' => 'ffffff',
	) ) );

	// Enable support for HTML5 markup.
	add_theme_support( 'html5', array( 'comment-list', 'search-form', 'comment-form', ) );
}
endif; // hemingway_metro_setup
add_action( 'after_setup_theme', 'hemingway_metro_setup' );

/**
 * Register widgetized area and update sidebar with default widgets.
 */
function hemingway_metro_widgets_init() {
	register_sidebar( array(
		'name'          => __( 'Sidebar', 'hemingway-metro' ),
		'id'            => 'sidebar-1',
		'before_widget' => '<aside id="%1$s" class="widget %2$s">',
		'after_widget'  => '</aside>',
		'before_title'  => '<h1 class="widget-title">',
		'after_title'   => '</h1>',
	) );
	register_sidebar( array(
		'name'          => __( 'Footer 1', 'hemingway-metro' ),
		'id'            => 'sidebar-2',
		'before_widget' => '<aside id="%1$s" class="widget %2$s">',
		'after_widget'  => '</aside>',
		'before_title'  => '<h1 class="widget-title">',
		'after_title'   => '</h1>',
	) );
	register_sidebar( array(
		'name'          => __( 'Footer 2', 'hemingway-metro' ),
		'id'            => 'sidebar-3',
		'before_widget' => '<aside id="%1$s" class="widget %2$s">',
		'after_widget'  => '</aside>',
		'before_title'  => '<h1 class="widget-title">',
		'after_title'   => '</h1>',
	) );
	register_sidebar( array(
		'name'          => __( 'Footer 3', 'hemingway-metro' ),
		'id'            => 'sidebar-4',
		'before_widget' => '<aside id="%1$s" class="widget %2$s">',
		'after_widget'  => '</aside>',
		'before_title'  => '<h1 class="widget-title">',
		'after_title'   => '</h1>',
	) );
}
add_action( 'widgets_init', 'hemingway_metro_widgets_init' );

/**
 * Enqueue scripts and styles.
 */
function hemingway_metro_scripts() {
	wp_enqueue_style( 'hemingway-metro-raleway' );
	wp_enqueue_style( 'hemingway-metro-latos' );
    wp_enqueue_style( 'flexslider-style', get_template_directory_uri() . '/inc/slider/css/flexslider.css' );
	wp_enqueue_style( 'hemingway-metro-style', get_stylesheet_uri() );
	wp_enqueue_style( 'genericons', get_template_directory_uri() . '/genericons/genericons.css' );

    wp_enqueue_script( 'jquery' );
    wp_enqueue_script( 'hemingway-metro-script', get_template_directory_uri() . '/js/hemingway-metro.js', array( 'jquery' ), '20140228', true );
    wp_enqueue_script( 'flexslider-script', get_template_directory_uri() .  '/inc/slider/js/jquery.flexslider-min.js', array( 'jquery' ), 'null', true );
    wp_enqueue_script( 'flexslider-init', get_template_directory_uri() .  '/inc/slider/js/flexslider-init.js', array( 'jquery' ), 'null', true );
	wp_enqueue_script( 'hemingway-metro-navigation', get_template_directory_uri() . '/js/navigation.js', array(), '20120206', true );
	wp_enqueue_script( 'hemingway-metro-skip-link-focus-fix', get_template_directory_uri() . '/js/skip-link-focus-fix.js', array(), '20130115', true );

	if ( is_singular() && comments_open() && get_option( 'thread_comments' ) ) {
		wp_enqueue_script( 'comment-reply' );
	}
}
add_action( 'wp_enqueue_scripts', 'hemingway_metro_scripts' );

/**
 * Register Google Fonts
 */
function hemingway_metro_google_fonts() {

	$protocol = is_ssl() ? 'https' : 'http';

	/*	translators: If there are characters in your language that are not supported
		by Raleway, translate this to 'off'. Do not translate into your own language. */

	if ( 'off' !== _x( 'on', 'Raleway font: on or off', 'hemingway-metro' ) ) {

		wp_register_style( 'hemingway-metro-raleway', "$protocol://fonts.googleapis.com/css?family=Raleway:400,300,700" );

	}

	/*	translators: If there are characters in your language that are not supported
		by Latos, translate this to 'off'. Do not translate into your own language. */

	if ( 'off' !== _x( 'on', 'Latos font: on or off', 'hemingway-metro' ) ) {

		wp_register_style( 'hemingway-metro-latos', "$protocol://fonts.googleapis.com/css?family=Lato:400,700,400italic,700italic" );

	}

}
add_action( 'init', 'hemingway_metro_google_fonts' );

/**
 * Enqueue Google Fonts for custom headers
 */
function hemingway_metro_admin_scripts( $hook_suffix ) {

	if ( 'appearance_page_custom-header' != $hook_suffix )
		return;

	wp_enqueue_style( 'hemingway-metro-raleway' );
	wp_enqueue_style( 'hemingway-metro-latos' );

}
add_action( 'admin_enqueue_scripts', 'hemingway_metro_admin_scripts' );

/**
 * Implement the Custom Header feature.
 */
require get_template_directory() . '/inc/custom-header.php';

/**
 * Custom template tags for this theme.
 */
require get_template_directory() . '/inc/template-tags.php';

/**
 * Custom functions that act independently of the theme templates.
 */
require get_template_directory() . '/inc/extras.php';

/**
 * Customizer additions.
 */
require get_template_directory() . '/inc/customizer.php';

/**
 * Load Jetpack compatibility file.
 */
require get_template_directory() . '/inc/jetpack.php';

/**
 * Create Slider Post Type
 */
require( get_stylesheet_directory() . '/inc/slider/slider_post_type.php' );

/**
 * Create Slider
 */
require( get_stylesheet_directory() . '/inc/slider/slider.php' );

/**
 * Adds "Read more link" for custom & automatic excerpts
 */
function hemingway_metro_read_more_link($output) {
    global $post;
    return $output . "<p><a class='more-link' href='". get_permalink() ."'>Continue reading <span class='meta-nav'>&rarr;</span></a></p>";
}

add_filter('the_excerpt', 'hemingway_metro_read_more_link');

/**
 * Removes the "[]" in automatic excerpts
 */
function hemingway_metro_excerpt_more( $more ) {
    return '…';
}
add_filter('excerpt_more', 'hemingway_metro_excerpt_more');
